### PR TITLE
Unify action checks

### PR DESCRIPTION
**Pull request by pablobm**
Fixes: https://github.com/thoughtbot/administrate/issues/1861

When referring to a route in the code, we run two checks:

* `valid_action?` is `true` if the route is defined, `false` otherwise.
* `show_action?` is expected to be overridden by developers in order to implement authorization. For example, it's implemented by `Administrate::Punditize` in order to integrate Administrate with Pundit. It should return `true` if the current user can access a given route, `false` otherwise.

These two check should (almost) always happen together. For this reason, our code is peppered with `if` statements where both are checked... and a few others where we forget one or the other.

These checks should be unified into a single method call, in order to avoid issues like the one described at https://github.com/thoughtbot/administrate/issues/1861. Here I propose such a method, which I call `accessible_action?`.

The original methods should still exist, as they do have their uses individually. The new method will delegate to the existing ones.

I have also renamed the two existing methods to something that I hope will make their intent clear:
- `valid_action?` becomes `existing_action?`
- `show_action?` becomes `authorized_action?`
- In order to provide a clear upgrade path, the old names still exist and work, but they show a deprecation warning when used. They can be removed properly at a later version of Administrate.

### Open question!
The original methods have a signature `the_old_question?(action, resource)`. I have flipped the order of parameters to be `the_new_question?(resource, action)`, as I find that more natural. But if people think the original order is better (or have some other proposal), I'm happy to oblige.

Also! The old `valid_action?` has a default parameter value (the full signature is `valid_action?(name, resource = resource_class)`). The new version doesn't have it, as I can't see that it has a good use case. Incidentally, this makes sense with the parameter order flip just described.

### CODE REVIEW

File: app/views/administrate/application/_collection.html.erb [hunk 1]
```diff
@@ -45,23 +45,21 @@ to display a collection of resources in an HTML table.
           <% end %>
         </th>
       <% end %>
-      <% [valid_action?(:edit, collection_presenter.resource_name),
-          valid_action?(:destroy, collection_presenter.resource_name)].count(true).times do %>
-        <th scope="col"></th>
-      <% end %>
```
Comment (c4lliope) [inline app/views/administrate/application/_collection.html.erb:51]: I'm glad you simplified this; no need to make things complex when we can render simple blank columns.

**Review by c4lliope — APPROVED** [review]
This change looks good; for your checklist, maybe you could rename:
* `valid_action?` -> `possible_action?`
* `show_action?` -> `permissible_action?`

...though I'm unsure because I'm barely involved here.
I'm sure you'll choose sensible names.

Much obliged for your changes, @pablobm !

**Comment by pablobm** [comment]
@c4lliope @nickcharlton - After more than a year, this PR is finally out of draft!

**Review by nickcharlton — APPROVED** [review]
🎉

**Pull request merged**